primap2/pm2io/_data_reading.py

```diff
@@ -1244,17 +1310,150 @@ def rename_columns(
     return attrs
 
 
+_special_codes = {
+    "C": np.nan,
+    "NaN": np.nan,
+    "nan": np.nan,
+    "-": 0,
+    "NE0": np.nan,
+    "": np.nan,
+}
+
+
+def is_float(to_test: Any) -> bool:
+    try:
+        float(to_test)
+        return True
+    except ValueError:
+        return False
+
+
+def find_str_values_in_data(
+    data: pd.DataFrame,
+    columns: List[str],
+) -> list:
+    """Find all string values occurring in given columns of a DataFrame"""
+    # limit our analysis to columns that contain strings
+    # (or other object types)
+    cols_with_strs = (
+        data[columns].select_dtypes(include=[object]).columns.values.tolist()
+    )
```

Comment:
What if a column is entirely string? Then it could also be of type `StringDtype`, right?

Reply:
I thought pandas always uses object when the data contains strings.